crates/anvil/src/eth/api.rs

```diff
@@ -941,32 +923,29 @@ impl EthApi {
             // but EIP-1559 prepends a version byte, so we need to encode the data first to get a
             // valid rlp and then rlp decode impl of `TypedTransaction` will remove and check the
             // version byte
-            let extend = rlp::encode(&data);
-            let tx = match rlp::decode::<TypedTransaction>(&extend[..]) {
+            let extend = alloy_rlp::encode(data);
+            let tx = match <TypedTransaction as alloy_rlp::Decodable>::decode(&mut &extend[..]) {
```

Comment:
does this not work?

```suggestion
            let tx = match TypedTransaction::decode(&mut &extend[..]) {
```

Reply:
https://github.com/foundry-rs/foundry/pull/6808/commits/51a4ada3af7a78dbde0e46dce5e13add2f20ceb9